### PR TITLE
nixos/release.nix: Refactoring for better multi-system support

### PR DESCRIPTION
**Pull request by dezgeg**
Currently, even if you pass `supportedSystems = [ "aarch64-linux" ]` you
end up with e.g. `nixos.iso_graphical.x86_64-linux` job. Using the new
forSubsetOfSystems avoids that.

This shouldn't affect the usual x86 trunk-combined jobset.

https://hydra.nixos.org/eval/1425558#tabs-removed for the effect on an aarch64-only jobset.

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]

File: nixos/release.nix [hunk 1]
```diff
@@ -13,6 +13,8 @@ let
 
   forAllSystems = genAttrs supportedSystems;
 
+  forSubsetOfSystems = systems: genAttrs (intersectLists supportedSystems systems);
```
Comment (Ericson2314) [inline nixos/release.nix]: Should this go in release-lib.nix?

Comment (dezgeg) [reply nixos/release.nix]: Uff... There's apparently already an equivalent called `forAllSupportedSystems` in release-lib.nix, but nixos/release.nix is not importing release-lib.nix

Comment (Ericson2314) [reply nixos/release.nix]: I'd a) import it, b) move `forAllSystems` to release-lib c) use `forAllSupportedSystems`.

Comment (dezgeg) [reply nixos/release.nix]: Yeah that seems the best (and `forAllSystems` could be implemented in terms of `forAllSupportedSystems` to reduce duplication). 

I hate the name `forAllSupportedSystems` though... given that we have `forAllSystems` which does something for every system passed in `supportedSystems`... and then `forAllSupportedSystems` which does something else.

Comment (Ericson2314) [reply nixos/release.nix]: I have no problem with you renaming them. Maybe something like `forAllSystems` and `forTheseSystems`?

**Comment by dezgeg** [comment]
Implemented the above suggestions.

**Comment by Ericson2314** [comment]
Glad you liked them, and thanks for making them!

**Pull request merged**